Model/GuestCart/PaymentUrl.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace MultiSafepay\Connect\Model\GuestCart;
+
+class PaymentUrl implements \MultiSafepay\Connect\Api\GuestPaymentUrlInterface
+{
+    /**
+     * @var \MultiSafepay\Connect\Model\PaymentUrl
+     */
+    protected $paymentUrlModel;
+
+    /**
+     * @var \Magento\Quote\Model\QuoteIdMaskFactory
+     */
+    protected $quoteIdMaskFactory;
+
+    /**
+     * PaymentUrl constructor.
+     * @param \MultiSafepay\Connect\Model\PaymentUrl $paymentUrlModel
+     * @param \Magento\Quote\Model\QuoteIdMaskFactory $quoteIdMaskFactory
+     */
+    public function __construct(
+        \MultiSafepay\Connect\Model\PaymentUrl $paymentUrlModel,
+        \Magento\Quote\Model\QuoteIdMaskFactory $quoteIdMaskFactory
+    ) {
+        $this->paymentUrlModel = $paymentUrlModel;
+        $this->quoteIdMaskFactory = $quoteIdMaskFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPaymentUrl($orderId, $cartId)
+    {
+        $quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
+
+        return $this->paymentUrlModel->getPaymentUrl($orderId, false, $quoteIdMask->getQuoteId());
```

Comment:
It is possible to use an invalid Quote_Masked_Id. If this is being used, you can recieve the payment link with just the order id. because of privacy reasons we can't merge this Pull request until the payment link can be recieved with the Quote_Masked Id and order Id. 

This can be confirmed by reaching the URL:
```https://example.com/rest/V1/multisafepay/invalid_value/payment-url/{{orderId}}```

I think the solution is to throw an Exception if ```$quoteIdMask->getQuoteId() === null```

Reply:
Nice find. We are testing this api today in our PWA. I will update the pull request